app/plugin.go

```diff
@@ -456,20 +460,24 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 func (a *App) GetPluginPublicKeys() ([]*model.PublicKeyDescription, *model.AppError) {
 	config := a.Config().PluginSettings
 
-	return config.PublicKeys, nil
+	return config.SignaturePublicKeyFiles, nil
 }
 
 // GetPublicKey will return the actual public key saved in the `filename` file.
 func (a *App) GetPublicKey(filename string) ([]byte, *model.AppError) {
+	ext := filepath.Ext(filename)
+	if PublicKeyFileExtention != ext {
```

Comment:
nit: this reads backwards to me (`expected != actual`, vs. `actual != expected`):
```suggestion
	if ext != PublicKeyFileExtention {
```